flow/layers/image_filter_layer.cc

```diff
@@ -29,6 +34,17 @@ void ImageFilterLayer::Preroll(PrerollContext* context,
   }
   set_paint_bounds(child_bounds);
 
+  if (context->damage_context) {
+    // Adjust display bounds for newly added layers
+    context->damage_context->ApplyImageFilter(
+        count, context->damage_context->layer_entries_count(), filter_.get(),
+        matrix, paint_bounds());
```

Comment:
Perhaps add a synthetic mutator to the DamageContext that performs these bounds adjustments on the fly?